src/main/java/com/google/firebase/auth/FirebaseUserManager.java

```diff
@@ -149,50 +148,31 @@ UserRecord getUserByPhoneNumber(String phoneNumber) throws FirebaseAuthException
   }
 
   String createUser(CreateRequest request) throws FirebaseAuthException {
-    GenericJson response;
-    try {
-      response = post("signupNewUser", request.getProperties(), GenericJson.class);
-    } catch (IOException e) {
-      throw new FirebaseAuthException(USER_CREATE_ERROR,
-          "IO error while creating user account", e);
-    }
-
+    GenericJson response = post(
+        "signupNewUser", request.getProperties(), GenericJson.class);
     if (response != null) {
       String uid = (String) response.get("localId");
       if (!Strings.isNullOrEmpty(uid)) {
         return uid;
       }
     }
-    throw new FirebaseAuthException(USER_CREATE_ERROR, "Failed to create new user");
+    throw new FirebaseAuthException(INTERNAL_ERROR, "Failed to create new user");
```

Comment:
If the UID is empty, is there indication of what is went wrong in the request body that we could show in this error?

Reply:
I don't think that situation actually occurs in this REST API. If something went wrong, it should respond with a different HTTP status. The check here is to prevent the SDK from returning a null/empty UID in the rare case the server misbehaves.